packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -173,6 +172,7 @@ export async function transformWithEsbuild(
   delete resolvedOptions.jsxInject
 
   try {
+    const { transform } = await import('esbuild')
```

Comment:
Calling dynamic import is a bit slow. If it's called in a hot path, it's better to cache the result: https://github.com/vitejs/vite-plugin-react/pull/141

Reply:
Ah, good call! Thanks Sapphi, we have a `importEsbuild` util now after https://github.com/vitejs/vite/pull/15621/commits/02f4bd5ce0367fd1197da4736752c225ee19252d